api/src/main/java/org/apache/iceberg/view/ViewVersion.java

```diff
@@ -43,8 +47,7 @@ public interface ViewVersion {
   long timestampMillis();
 
   /**
-   * Return the version summary such as the name of the operation that created that version of the
-   * view
+   * Return a string map of summary data for the operation that produced this view version.
```

Comment:
In the data model itself, I think we should follow what Snapshot does where summary returns all the fields other than the operation itself. We have a separate operation method below for the operation. When we serialize the version, the operation will get put in the summary of the version. Let me know if this makes sense @nastra @jackye1995 @rdblue

Reply:
I think we can avoid having a change of spec here, the information in summary are just FYI purpose, I don't think we need to assign a semantic meaning to operations now, we can always add later if it becomes necessary. But for views I don't see anything other than CREATE and REPLACE, but it can also be identified directly through the version ID, ID=1 means CREATE, everything else are REPLACE.

Reply:
Sorry for the confusion here, there's no change in spec here, it's a change in the original API layer. We'll still serialize the `operation` in the `summary` map when writing the metadata file, so the spec is maintained. 

When we deserialize into the in memory representation, the summary map will not contain `operation`, and instead a user would use the `operation()` API. It's a similar model to `Snapshot` and `SnapshotParser`

Let me know if that makes sense, or if there's doubts on the API itself after this change! @rdblue

Reply:
Discussed offline with @jackye1995 instead of restricting the type of operation up front, we can keep it open. It looks like Snapshot summary also does has the same behavior. There's a set of recognized operations by the Iceberg library like `append`, `replace` etc but those aren't strictly validated when  reading the metadata. So that's a similar model we can follow here. We also don't need an enum following this logic, it can just be defined Strings (which we can define separately when we get to the catalog implementation for `createView` `replaceView` etc. Also we can always add an `operation` API if it becomes to awkward to get the operation from the summary but it's not needed at the moment.

Reply:
Discussed offline with @jackye1995 we were talking on two different issues here. 

1.) The first is on strict validation of the `operation`, we concluded that's probably not a desired behavior, and that's the behavior of `snapshot` currently.

2.) The second is on the API and serialization of operation itself. Operation will be serialized in the summary but when it's read we deserialize it into a separate field in memory. This is the same as what happens with `Snapshot` and I think it makes sense to have that same behavior and have a separate API for operation. Updated the PR to reflect this.